turbopack/crates/turbo-tasks-backend/src/backend/operation/mod.rs

```diff
@@ -445,11 +451,47 @@ impl<B: BackingStorage> TaskGuard for TaskGuardImpl<'_, B> {
     }
 
     fn add_new(&mut self, item: CachedDataItem) {
-        self.check_access(item.category());
-        let added = self.add(item);
+        let category = item.category();
+        self.check_access(category);
+        if !self.task_id.is_transient() && item.is_persistent() {
+            self.task.track_modification(category.into_specific());
+        }
+        let added = self.task.add(item);
         assert!(added, "Item already exists");
```

Comment:
```suggestion
        let added = self.task.add(item);
        assert!(added, "Item already exists");
```

add already checks access and tracks so we can drop it here afaict

Reply:
It's not calling `self.add` but `self.task.add` which doesn't do these things.

The old code did check access twice.

I have chosen to not call `self.add` as I want to avoid the unnecessary `contains_key` for the `add_new` method, which is always a new item.

Reply:
Hey, on v16.0.0-canary.0 I'm constantly this fatal panic log "Turbopack Error: Item already exists" coming from this file on a slow file system. I wonder if when this merges it'll be fixed?